### ISSUE TITLE
Strange issue with dropbox tray icon rendering outside of bar

### ISSUE THREAD

**Issue by kristijanhusak** [polybar/polybar#2430]
## Checklist

* [X] I have read the appropriate section in the [contributing
  guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
* [X] I believe this issue is a problem with polybar itself and not a misconfiguration on my part.
* [X] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that
  may have already reported this problem.
* [X] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.

## Describe the bug

Few days ago my dropbox icon started to render outside of tray, on complete opposite side of screen.

**Expected behavior:**

I expect all tray icons to be in the same place where i define them.

**Actual behavior:**

Dropbox tray icon (and only dropbox) renders on the opposite side of screen, way outside of tray area where other tray icons live.

**Was it working before?**
* Did you also experience this bug in an earlier version of polybar (yes/no/don't know)? I don't know, started happening few days ago, not sure if polybar even updated or even if it's polybar issue.
* If no, what was the last version where this worked correctly?

## To Reproduce

Here's link to it: https://github.com/kristijanhusak/neovim-config/blob/bleeding-edge/i3/polybar/config
I didn't change config from November 2020, so it's definetely not an issue with config (was working before)

This issue happens to me on 3 machines, all running the same system (Manjaro i3). 2 Desktops and 1 laptop.

## Polybar Log

```
notice: Parsing config file: /home/kristijan/.config/polybar/config
error: Disabling module "battery" (reason: No suitable way to get current charge state)
error: Disabling module "wlan" (reason: basic_string::_M_construct null not valid)
notice: pulseaudio: using default sink alsa_output.pci-0000_00_1f.3.analog-stereo
notice: Loaded font "JetBrains Mono:style=Regular:pixelsize=9" (name=JetBrains Mono, offset=1, file=/usr/share/fonts/TTF/JetBrainsMono-Regular.ttf)
notice: Loaded font "RobotoMono Nerd Font Mono:style=Regular:pixelsize=9" (name=RobotoMono Nerd Font Mono, offset=1, file=/usr/share/fonts/TTF/Roboto Mono Nerd Font Complete Mono.ttf)
notice: Loaded font "Hasklig:style=Regular:pixelsize=9" (name=Hasklig, offset=1, file=/usr/share/fonts/OTF/Hasklig-Regular.otf)
notice: Loaded font "Noto Sans Symbols2:style=Regular:pixelsize=9" (name=Noto Sans Symbols2, offset=1, file=/usr/share/fonts/noto/NotoSansSymbols2-Regular.ttf)
warn: Ignoring restack of i3 window (not needed when `override-redirect = false`)
error: Error in X event loop: XCB_MATCH (8)
```

## Screenshots
Note the top left corner.
Other tray icons are in bottom right
![screenshot](https://i.imgur.com/bHg3urS.jpg)

## Environment:
* WM: i3
* Distro: Manjaro i3 Linux
* Output of `polybar -vvv`:

```
polybar 3.5.5-117-g735a08ba

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2,-D_GLIBCXX_ASSERTIONS         -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wall -Wextra -Wpedantic -Wsuggest-override -O3 -DNDEBUG
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 
```

## Additional context
I'm not sure if this is a polybar issue. I tried testing the default i3bar, and tray properly renders there (with dropbox icon properly inside) [proof](https://i.imgur.com/LR1QVko.jpg), which lead me to assumption that it's a polybar issue.

**Comment by podiki** [comment]
Same thing here, on polybar 3.5.5 (non git version), on XMonad. I'm guessing it was a change in Dropbox (or something it uses for tray icon) and its interaction with polybar? For me, also appears on the top left, though about one Dropbox-icon-width to the left. My tray is on the top right and everything else is there.

**Comment by kristijanhusak** [comment]
> EDIT: issuing polybar-msg cmd restart causes it to move in to the right a little bit each time (or at least for the first ~3 times). Quitting and restarting Dropbox doesn't help.

This also happens to me. It moves 3 times from top left to the right and stops.

**Comment by podiki** [comment]
Glad I'm not alone, and my guess is something Dropbox/its tray icon does. It has always been the odd one out for me, as in #854. (Another reason I should completely phase out Dropbox, but that's another matter...)

**Comment by patrick96** [comment]
I started experiencing this as well, though polybar just stopped displaying completely.

What happened was that polybar failed to configure the dropbox tray window for some reason and there was a bug in the code that did not remove the window when there was an error (this caused dropbox to be positioned outside of the bar).

The actual error when configuring the dropbox tray window was because of a mismatch in color depth. I think i have a fix for that.

**Comment by patrick96** [comment]
Alright, I have a fix in #2431. Could you please try it out and see if it 1) fixes the dropbox issue and 2) does not break anything else.

**Comment by kristijanhusak** [comment]
@patrick96 Just tried it, it works!

![screenshot](https://i.imgur.com/3Pn6u56.png)

**Comment by podiki** [comment]
Yes, #2431 works for me too, hooray! (Will keep any eye out for any regressions, but didn't notice any so far.)

**Comment by shahab-vahedi** [comment]
> Yes, #2431 works for me too, hooray! (Will keep any eye out for any regressions, but didn't notice any so far.)

@podiki Could you kindly look into this [issue](https://github.com/polybar/polybar/issues/2442)? I think there are some regression involved.

**Comment by stevensonmt** [comment]
Was this fix already released? I'm on 3.5.5-1 and have the dropbox icon floating all over the place instead of in the tray to the right.

**Comment by patrick96** [comment]
@stevensonmt Yes, it is in 3.5.6
